asyn/asynPortDriver/paramVal.h

```diff
@@ -31,8 +33,8 @@ class paramVal {
     epicsUInt32 getUInt32(epicsUInt32 valueMask);
     void setDouble(epicsFloat64 value);
     epicsFloat64 getDouble();
-    void setString(const char *value);
-    char *getString();
+    void setString(std::string value);
+    std::string getString();
```

Comment:
Be careful as there is potentially a string copy happening here.  Unless a returning a temporary, it is almost always preferable to return a const reference.

```
const std::string& getString();
```

Also note that changing return type between value and const reference generally doesn't require changes to API user code.